R/module_init_data.R

```diff
@@ -151,7 +154,7 @@ srv_init_data <- function(id, data, modules, filter = teal_slices()) {
       list(code = trimws(c(teal.code::get_code(data), hashes), which = "right")),
       list(join_keys = teal.data::join_keys(data)),
       sapply(
-        ls(teal.code::get_env(data)),
+        .teal_data_ls(data),
```

Comment:
why this new function?
If you are doing `ls()` this should return all the names - no need to explicitly append parents
you might want to reorder the objects but if we are in the environment then the order does not matter (which is a big advantage of the envir compared to list)
I think it should be fine to stick to `ls(..., all.names = FALSE)` and remove `.teal_data_ls()` at all

Reply:
> I think it should be fine to stick to ls(..., all.names = FALSE) and remove .teal_data_ls() at all

It is not enough. I've created `.teal_data_ls()` because it does topological sort - to display datasets in the filter panel.

Reply:
understood 
I totally got that this functionality must to be there but this is not very clear to me that this is the place responsible for such reorder. Maybe we should move it somewhere else closer to the filter panel
the description for this function is that this is responsible for signature protection

Reply:
Problem is that not only filter-panel displays datanames in topological order but also summary table (which is the part of teal now). I can move `.include_parent_datanames` away from `.teal_data_ls` but it will probably always go together. 

_`.include_parent_datanames` adds parents if missing and orders topologically_

Reply:
I've just have removed `.teal_data_ls` in favour of decoupled `ls()` + `.topologically_sort_datanames()`